diff -r -U 3 minecraft/net/minecraft/client/controller/PlayerControllerCreative.java minecraft_patched/net/minecraft/client/controller/PlayerControllerCreative.java
--- minecraft/net/minecraft/client/controller/PlayerControllerCreative.java	2021-11-26 19:31:50.996590400 +0200
+++ minecraft_patched/net/minecraft/client/controller/PlayerControllerCreative.java	2021-11-26 19:32:52.027081200 +0200
@@ -1,5 +1,6 @@
 package net.minecraft.client.controller;
 
+import net.minecraft.client.Minecraft;
 import net.minecraft.client.Session;
 import net.minecraft.game.entity.player.EntityPlayer;
 import net.minecraft.game.item.ItemStack;
@@ -8,6 +9,10 @@
 import net.minecraft.game.level.block.Block;
 
 public final class PlayerControllerCreative extends PlayerController {
+	public PlayerControllerCreative(Minecraft var1) {
+		super(var1);
+	}
+
 	private MobSpawner mobSpawner;
 
 	public final void onRespawn(EntityPlayer var1) {
diff -r -U 3 minecraft/net/minecraft/client/ThreadDownloadResources.java minecraft_patched/net/minecraft/client/ThreadDownloadResources.java
--- minecraft/net/minecraft/client/ThreadDownloadResources.java	2021-11-26 19:31:50.992590200 +0200
+++ minecraft_patched/net/minecraft/client/ThreadDownloadResources.java	2021-11-26 20:24:37.068679400 +0200
@@ -1,11 +1,14 @@
 package net.minecraft.client;
 
+import java.io.BufferedReader;
 import java.io.DataInputStream;
 import java.io.DataOutputStream;
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.io.InputStreamReader;
 import java.net.URL;
+import java.util.ArrayList;
 
 public final class ThreadDownloadResources extends Thread {
 	private File resourcesFolder;
@@ -23,7 +26,63 @@
 	}
 
 	public final void run() {
-		// $FF: Couldn't be decompiled
+		try {
+			final ArrayList<String> list = new ArrayList<String>();
+			final URL url = new URL("http://www.minecraft.net/resources/");
+			final BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(url.openStream()));
+			String line;
+			while ((line = bufferedReader.readLine()) != null) {
+				list.add(line);
+			}
+			bufferedReader.close();
+			for (int i = 0; i < list.size(); ++i) {
+				final URL url2 = url;
+				final String s = list.get(i);
+				final URL context = url2;
+				Label_0334: {
+					try {
+						final String[] split;
+						final String child = (split = s.split(","))[0];
+						final int int1 = Integer.parseInt(split[1]);
+						Long.parseLong(split[2]);
+						final File file;
+						if (!(file = new File(this.resourcesFolder, child)).exists() || file.length() != int1) {
+							file.getParentFile().mkdirs();
+							this.downloadResource(new URL(context, child.replaceAll(" ", "%20")), file);
+							if (this.closing) {
+								break Label_0334;
+							}
+						}
+						final Minecraft mc = this.mc;
+						final String s2 = child;
+						final File file2 = file;
+						final String s3 = s2;
+						final Minecraft minecraft = mc;
+						final int index = s3.indexOf("/");
+						final String substring = s3.substring(0, index);
+						final String substring2 = s3.substring(index + 1);
+						if (substring.equalsIgnoreCase("sound")) {
+							minecraft.sndManager.addSound(substring2, file2);
+						}
+						else if (substring.equalsIgnoreCase("newsound")) {
+							minecraft.sndManager.addSound(substring2, file2);
+						}
+						else if (substring.equalsIgnoreCase("music")) {
+							minecraft.sndManager.addMusic(substring2, file2);
+						}
+					}
+					catch (Exception ex) {
+						ex.printStackTrace();
+					}
+				}
+				if (this.closing) {
+					return;
+				}
+			}
+		}
+		catch (IOException ex2) {
+			ex2.printStackTrace();
+		}
 	}
 
 	private void downloadResource(URL var1, File var2) throws IOException {
diff -r -U 3 minecraft/net/minecraft/game/item/Item.java minecraft_patched/net/minecraft/game/item/Item.java
--- minecraft/net/minecraft/game/item/Item.java	2021-11-26 20:39:31.146817700 +0200
+++ minecraft_patched/net/minecraft/game/item/Item.java	2021-11-26 20:39:01.851142100 +0200
@@ -46,1 +46,1 @@  
-	private int iconIndex;
+	protected int iconIndex;


diff -r -U 3 minecraft/net/minecraft/client/model/md3/MD3Loader.java minecraft_patched/net/minecraft/client/model/md3/MD3Loader.java
--- minecraft/net/minecraft/client/model/md3/MD3Loader.java	2021-11-26 20:39:31.146817700 +0200
+++ minecraft_patched/net/minecraft/client/model/md3/MD3Loader.java	2021-11-26 20:39:01.851142100 +0200
@@ -12,1 +12,1 @@  
-	public final MD3Vertices loadModel(String var1) {
+	public final MD3Vertices loadModel(String var1) throws IOException {
@@ -28,1 +28,1 @@  
-	private MD3Vertices loadMD3Data(ByteBuffer var1) {
+	private MD3Vertices loadMD3Data(ByteBuffer var1) throws IOException {
@@ -95,1 +95,1 @@  
-	private MD3Buffers getMD3Buffer(ByteBuffer var1) {
+	private MD3Buffers getMD3Buffer(ByteBuffer var1) throws IOException {